tests/integration_tests/test_experiment.py

```diff
@@ -252,66 +253,64 @@ def test_experiment_multiple_seq_seq(csv_filename, output_features):
 @pytest.mark.parametrize("image_source", ["file", "tensor"])
 @pytest.mark.parametrize("num_channels", [1, 3])
 def test_basic_image_feature(num_channels, image_source, in_memory, skip_save_processed_input, csv_filename):
-    # Image Inputs
-    image_dest_folder = os.path.join(os.getcwd(), "generated_images")
-
-    input_features = [
-        image_feature(
-            folder=image_dest_folder,
-            encoder="stacked_cnn",
-            preprocessing={
-                "in_memory": in_memory,
-                "height": 12,
-                "width": 12,
-                "num_channels": num_channels,
-                "num_processes": 5,
-            },
-            fc_size=16,
-            num_filters=8,
-        )
-    ]
-    output_features = [category_feature(vocab_size=2, reduce_input="sum")]
-
-    rel_path = generate_data(input_features, output_features, csv_filename)
-
-    if image_source == "file":
-        # use images from file
-        run_experiment(
-            input_features, output_features, dataset=rel_path, skip_save_processed_input=skip_save_processed_input
-        )
-    else:
-        # import image from file and store in dataframe as tensors.
-        df = pd.read_csv(rel_path)
-        image_feature_name = input_features[0]["name"]
-        df[image_feature_name] = df[image_feature_name].apply(lambda x: torchvision.io.read_image(x))
-
-        run_experiment(input_features, output_features, dataset=df, skip_save_processed_input=skip_save_processed_input)
-
-    # Delete the temporary data created
-    shutil.rmtree(image_dest_folder, ignore_errors=True)
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
`tmpdir` can also be a fixture you pass in as a param to the test function, so you don't need to use a context manager here.

Reply:
Thanks for the suggestion! Done.